src/modules/m_lockserv.cpp

```diff
@@ -27,23 +27,23 @@
 
 class CommandLockserv : public Command
 {
-	bool& locked;
+	std::string& locked;
 
  public:
-	CommandLockserv(Module* Creator, bool& lock) : Command(Creator, "LOCKSERV", 0), locked(lock)
+	CommandLockserv(Module* Creator, std::string& lock) : Command(Creator, "LOCKSERV", 0), locked(lock)
 	{
 		flags_needed = 'o';
 	}
 
-	CmdResult Handle (const std::vector<std::string> &parameters, User *user)
+	CmdResult Handle(const std::vector<std::string>& parameters, User* user)
 	{
-		if (locked)
+		if (!locked.empty())
 		{
 			user->WriteNotice("The server is already locked.");
 			return CMD_FAILURE;
 		}
 
-		locked = true;
+		locked = parameters.empty() || parameters[0].empty() ? "Server is temporarily closed. Please try again later." : parameters[0];
```

Comment:
I switched to using `allow_empty_last_param = false` so the second check is unnecessary